src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -913,6 +922,114 @@ public Page search(String q) throws Exception {
     }
 
     /**
+     * Get JSON from A Jenkins endpoint.
+     * @param path The endpoint URL.
+     * @return The JSON.
+     */
+    public JSONWebResponse getJSON(@Nonnull String path) throws IOException {
+        assert !path.startsWith("/");
+
+        JenkinsRule.WebClient webClient = createWebClient();
+        Page runsPage = null;
+        try {
+            runsPage = webClient.goTo(path, "application/json");
+        } catch (SAXException e) {
+            // goTo shouldn't be throwing a SAXException for JSON.
+            throw new IllegalStateException("Unexpected SAXException.", e);
+        }
+        WebResponse webResponse = runsPage.getWebResponse();
+
+        return new JSONWebResponse(webResponse);
+    }
+
+    /**
+     * POST a JSON payload to a URL on the underlying Jenkins instance.
+     * @param path The url path on Jenkins.
+     * @param json An object that produces a JSON string from it's {@code toString} method.
+     * @return A JSON response.
+     * @throws IOException
+     * @throws SAXException
+     */
+    public JSONWebResponse postJSON(@Nonnull String path, @Nonnull Object json) throws IOException, SAXException {
+        assert !path.startsWith("/");
+
+        URL postUrl = new URL(getURL().toExternalForm() + path);
```

Comment:
[Yes](https://github.com/tfennelly/jenkins-test-harness/blob/dfac492d95060e1ad41c75c425cf09982d0b5c9e/src/main/java/org/jvnet/hudson/test/JenkinsRule.java#L798).

BTW `toString()` is an alias for `toExternalForm()` so `getURL() + path` suffices.